Umbraco Commerce v15 updates (#6637)
umbraco/UmbracoDocs: 15/umbraco-commerce/key-concepts/payment-providers.md

Generally, there are three methods within a Payment Provider that you may need to implement, and each one has a specific responsibility.

> * **GenerateFormAsync** - The `GenerateFormAsync` method is responsible for generating an HTML form that will redirect the customer to the given payment gateway payment form. In this method you may need to communicate with the payment gateway in order to initialize a payment, letting the payment gateway know how much to capture. This often results in some kind of code or redirect URL being returned which will need to be embedded into the generated form. The generated form is then usually displayed on a checkout **Review** page, the last page before payment is captured and will have an implementer-defined **Continue to Payment** button to submit the form and redirect the customer to the gateway.

Reviewer: ```suggestion
* **GenerateFormAsync** - The `GenerateFormAsync` method generates an HTML form that will redirect the customer to the given payment gateway payment form. In this method you may need to communicate with the payment gateway to initialize a payment, letting the payment gateway know how much to capture. This often results in some kind of code or redirect URL being returned which will need to be embedded into the generated form. The form appears on a checkout **Review** page before payment, featuring a **Continue to Payment** button to submit the form and redirect to the gateway.
```

Reviewer: shortened long sentence.

> * **ProcessCallbackAsync** - The `ProcessCallbackAsync` method is responsible for handling the response coming back from the payment gateway and processing whether the payment was successful or not. This can sometimes occur _synchronously_, if the payment gateway sends information back as part of the confirmation page redirect, or can occur _asynchronously_ if the payment gateway sends the information back via an out-of-band webhook request.

Reviewer: ```suggestion
* **ProcessCallbackAsync** - The `ProcessCallbackAsync` method handles the response coming back from the payment gateway and processing whether the payment was successful or not. This can occur _synchronously_, if the payment gateway sends information back during the confirmation page redirect. It can also occur _asynchronously_ if the payment gateway sends the information back via an out-of-band webhook request.
```

> * **GetOrderReferenceAsync** - The `GetOrderReferenceAsync` method is responsible for extracting an order reference number from a request when the payment gateway uses an asynchronous webhook to finalize an Order **and** it uses a global webhook URL strategy for all notifications rather than a notification URL per transaction. Where a webhook URL can be passed per transaction, then Umbraco Commerce provides you with a unique callback URL you can register with the gateway that already identifies the order reference as part of the URL parameters, making implementing this method unnecessary.

Reviewer: ```suggestion
* **GetOrderReferenceAsync** - The `GetOrderReferenceAsync` method extracts an order reference number from a request when a payment gateway uses an asynchronous webhook to finalize an order.  This method is specifically used where the payment gateway uses a global webhook URL strategy for all notifications rather than a notification URL per transaction. Umbraco Commerce provides a unique callback URL for each transaction, which includes the order reference as part of the URL parameters. This makes it unnecessary to pass a webhook URL.
```

Author: Maybe change the last bit to be

> Where a webhook URL can be passed per transaction, then Umbraco Commerce provides you with a unique callback URL that already identifies the order reference as part of the URL parameters. When using the per transaction URL provided, implementing this method becomes unnecessary.
